src/codec/decode.rs

```diff
@@ -1,13 +1,39 @@
 use super::types::*;
 use num_bigint::BigInt;
 use num_traits::ToPrimitive;
+
+#[cfg(feature = "std")]
 use std::collections::BTreeMap;
+#[cfg(feature = "std")]
 use std::error::Error;
+#[cfg(feature = "std")]
 use std::fmt;
+#[cfg(feature = "std")]
 use std::result::Result;
+#[cfg(feature = "std")]
 use std::str;
+#[cfg(feature = "std")]
 use std::str::FromStr;
 
+#[cfg(not(feature = "std"))]
+use alloc::collections::BTreeMap;
+#[cfg(not(feature = "std"))]
+use alloc::string::{String, ToString};
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+#[cfg(not(feature = "std"))]
+use core::fmt;
+#[cfg(not(feature = "std"))]
+use core::result::Result;
+#[cfg(not(feature = "std"))]
+use core::result::Result;
+#[cfg(not(feature = "std"))]
+use core::str;
+#[cfg(not(feature = "std"))]
+use core::str;
+#[cfg(not(feature = "std"))]
+use core::str::FromStr;
+
```

Comment:
Use `alloc` and `core` only if possible because some `std::*` is just re-export of `alloc` and `core`.